### PR TITLE
Added basic documentation, secret version support, and mock test

### PR DESCRIPTION
**Pull request by jtaubensee**
## Description
* Added support for secret version when creating a new `KeyVaultPlugin` instance
* Added a mock test and created `ISecretManager` interface
* Added basic documentation

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](./CONTRIBUTING.md).**
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR.
- [x] The pull request does not introduce breaking changes (unless a major version change occurs in the assembly and module).
- [x] If applicable, the public code is properly documented.
- [x] Pull request includes test coverage for the included changes.
- [x] The code builds without any errors.

### CODE REVIEW

File: docs/keyvault.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Key Vault plugin for Azure Service Bus
+
+The Key Vault plugin for Azure Service Bus allows for the message body to be encrypted and decrypted by simply registering a plugin.
```
Comment (SeanFeldman) [inline docs/keyvault.md]: _and_ keyvault settings?

File: src/Microsoft.Azure.ServiceBus.KeyVault/KeyVaultSecretManager.cs [hunk 2]
```diff
@@ -42,67 +43,71 @@ internal KeyVaultSecretManager(string keyVaultUrl, string azureClientId, string
...
             }
 
-            var secret = await GetSecretFromKeyVault(secretName);
+            var secret = await GetSecretFromKeyVault(secretName, secretVersion);
             using (var sha256 = SHA256.Create())
             {
                 var secretAsBytes = Encoding.UTF8.GetBytes(secret);
                 var hashedSecret = sha256.ComputeHash(secretAsBytes);
-                secretCache.Add(secretName, hashedSecret);
+                secretCache.Add(combinedNameAndVersion, hashedSecret);
```
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus.KeyVault/KeyVaultSecretManager.cs]: Considering that the code can be executed concurrently, should `secretCache` be a `ConcurrentDictionary`?

File: src/Microsoft.Azure.ServiceBus.KeyVault/KeyVaultSecretManager.cs [hunk 3]
```diff
@@ -36,73 +37,77 @@ internal KeyVaultSecretManager(string keyVaultUrl, string azureClientId, string
...
+                    }
+                    else
+                    {
+                        secretResult = await keyVaultClient.GetSecretAsync(this.KeyVaultUrl, secretName, secretVersion);
+                    }
+                    return secretResult.Value;
                 }
                 catch (Exception ex)
                 {
                     throw new KeyVaultPluginException(string.Format(Resources.KeyVaultKeyAcquisitionFailure, secretName, KeyVaultUrl), ex);
```
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus.KeyVault/KeyVaultSecretManager.cs]: >throw new KeyVaultPluginException(string.Format(Resources.KeyVaultKeyAcquisitionFailure, secretName, KeyVaultUrl), ex); [](start = 20, length = 119)

update this with version info as well

File: src/Microsoft.Azure.ServiceBus.KeyVault/KeyVaultPlugin.cs [hunk 4]
```diff
@@ -31,6 +31,17 @@ public class KeyVaultPlugin : ServiceBusPlugin
...
+        {
+        }
+
+        /// <summary>
+        /// Creates a new instance of an <see cref="KeyVaultPlugin"/>.
+        /// </summary>
+        /// <param name="encryptionSecretName">The name of the secret used to encrypt / decrypt messages.</param>
+        /// <param name="encryptionSecretVersion">The version of the secret</param>
+        /// <param name="options">The <see cref="KeyVaultPluginSettings"/> used to create a new instance.</param>
+        public KeyVaultPlugin(string encryptionSecretName, string encryptionSecretVersion, KeyVaultPluginSettings options)
```
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus.KeyVault/KeyVaultPlugin.cs]: >options [](start = 114, length = 7)

rename this param to keyVaultSettings.
Also, update the name in docxml.

File: docs/keyvault.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+The Key Vault plugin for Azure Service Bus allows for the message body to be encrypted and decrypted by registering a plugin and its accompanying settings.
+
+## How to use
+
+In order to use this plugin you will need to setup the following:
+
+1. An Azure subscription
+
+1. A Service Bus namespace
```
Comment (nemakam) [inline docs/keyvault.md:11]: >1 [](start = 0, length = 1)

Increment the numbering? :)

**Review by nemakam — APPROVED** [review]

**Comment by nemakam** [comment]
@jtaubensee , I've approved the PR. Please take care of the nit comments and merge it

File: docs/keyvault.md (hunk 5 above)
Comment (SeanFeldman) [reply docs/keyvault.md:11]: No need. Remove the empty lines in between to have a proper numbered list GH will respect.

**Comment by nemakam** [comment]
:shipit:

**Comment by SeanFeldman** [comment]
Who's shipping this? 😃

File: docs/keyvault.md (hunk 5 above)
Comment (jtaubensee) [reply docs/keyvault.md:11]: This renders properly as is

Comment (SeanFeldman) [reply docs/keyvault.md:11]: GH is 🤓

**Pull request merged**